clients/src/main/java/org/apache/kafka/clients/consumer/KafkaShareConsumer.java

```diff
@@ -115,23 +115,26 @@
  * as naturally happens when the locks time out. This limit is controlled by the broker configuration property
  * {@code group.share.record.lock.partition.limit}. By limiting the duration of the acquisition lock and automatically
  * releasing the locks, the broker ensures delivery progresses even in the presence of consumer failures.
+ *
  * <p>
- * The consumer can choose to use implicit or explicit acknowledgement of the records it processes.
- * <p>If the application calls {@link #acknowledge(ConsumerRecord, AcknowledgeType)} for any record in the batch,
- * it is using <em>explicit acknowledgement</em>. In this case:
+ * The consumer can choose to use implicit or explicit acknowledgement of the records it processes by configuring the
+ * {@link ConsumerConfig#SHARE_ACKNOWLEDGEMENT_MODE_CONFIG} property. If the property is not set, the default mode is <code>"implicit"</code>.
+ *
+ * <p>If the config is set to "explicit", the consumer is using <em>explicit acknowledgement</em>. In this case:
  * <ul>
+ *     <li>The application is expected to acknowledge all the records it received in the batch before the next call to ({@link #poll(Duration)}</li>
```

Comment:
Let's make it stronger. `The application must acknowledge all the records it received in the batch before`... Also, the line lengths are intentionally quite short here for readability so please split this line.